processor.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+var doiuse = require('doiuse');
+var github = require('octonode');
+var postcss = require('postcss');
+var sourceMap = require('source-map');
+var stylus = require('stylus');
+var logger = require('./logger');
+
+function processCSS(originRepo, originBranch, file, config, handleIncompatibility) {
+    var repoClient = github.client().repo(originRepo);
+
+    repoClient.contents(file.filename, originBranch, function(error, fileContentsMetadata) {
+        var fileContents;
+
+        if (error) return logger.error('Error reading CSS file contents:', error);
+
+        fileContents = new Buffer(fileContentsMetadata.content, 'base64').toString();
+
+        postcss(doiuse({
+            browsers: config,
+            onFeatureUsage: handleIncompatibility
+        })).process(fileContents, {
+            from: '/' + file.filename
+        }).then(function(response) {});
+    });
+}
+
+function processStylus(originRepo, originBranch, file, config, handleIncompatibility) {
+    var repoClient = github.client().repo(originRepo);
+
+    repoClient.contents(file.filename, originBranch, function(error, fileContentsMetadata) {
+        var fileContents, compiler;
+
+        if (error) return logger.error('Error reading Stylus file contents:', error);
+
+        fileContents = new Buffer(fileContentsMetadata.content, 'base64').toString();
+        compiler = stylus(fileContents)
+            .set('filename', file.filename)
+            .set('sourcemap', {});
+
+        compiler.render(function(error, compiledCSS) {
+            var sourceMapConsumer;
+
+            if (error) return logger.error('Error compiling Stylus:', error);
+
+            sourceMapConsumer = new sourceMap.SourceMapConsumer(compiler.sourcemap);
+
+            function handleStylusIncompatibility(incompatibility) {
+                var compiledPosition = sourceMapConsumer.originalPositionFor(incompatibility.usage.source.start);
```

Comment:
Since we use `incompatibility.usage.source.start` in 3 lines, we should create a shortcut reference var for it.